src/analysis/special_functions/exp_log.lean

```diff
@@ -486,6 +486,15 @@ begin
   rwa [abs_of_neg hy, abs_of_neg hx, neg_lt_neg_iff]
 end
 
+lemma log_inj_on_pos : set.inj_on log (set.Ioi 0) :=
+strict_mono_incr_on_log.inj_on
+
+lemma one_of_pos_of_log_eq_zero {x : ℝ} (h₁ : 0 < x) (h₂ : log x = 0) : x = 1 :=
+log_inj_on_pos (set.mem_Ioi.2 h₁) (set.mem_Ioi.2 zero_lt_one) (h₂.trans real.log_one.symm)
+
+lemma log_nonzero_of_ne_one (x : ℝ) (hx_pos : 0 < x) (hx : x ≠ 1) : log x ≠ 0 :=
```

Comment:
Shouldn't this be `log_ne_zero_of_ne_one`? Also, I think `x` should be implicit here.

Reply:
Good point, `x` should be implicit, and yes, in this case "ne_zero" seems more appropriate than "nonzero"